src/shared/tests.h

```diff
@@ -3,6 +3,27 @@
 
 #include <stdbool.h>
 
+#include "sd-daemon.h"
+
+#include "string-util.h"
+
+#define TEST_REQ_RUNNING_SYSTEMD(x)                                 \
+        if (sd_booted() > 0) {                                      \
+                x;                                                  \
+        } else {                                                    \
+                printf("systemd not booted skipping '%s'\n", #x);   \
+        }
```

Comment:
hmm, i'd really enclose this in `do {` and `} while(false)` , so that any trailing `;` isn't considered weird...